Training24/Test.c

```diff
@@ -1,14 +1,151 @@
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+static void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+static void DisplayArray (int arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i]);
+   printf ("|\n");
+}
+
+/// <summary>Function to check if the array is sorted.</summary>
+static bool IsSorted (int arr[], int size) {
+   for (int i = 1; i < size; i++) if (arr[i - 1] > arr[i]) return false;
+   return true;
+}
+
+/// <summary>Checks if the input string is a valid integer (digits only).</summary>
+static int IsValidInteger (const char* str) {
+   if (str == NULL || *str == '\n' || *str == '\0') return 0;
+   while (isspace (*str)) str++;
+   if (*str == '+' || *str == '-') str++;
+   for (; *str != '\0' && *str != '\n'; str++) if (!isdigit (*str)) return 0;
+   return 1;
+}
+
+/// <summary>Function to format and print test results.</summary>
+static void PrintTestCase (int numTests, int inputs[], int expected[], int sortedArray[], int size, int searchElement, int index) {
+   printf (YELLOW "------------------ Test Case %d -------------------\n" RESET, numTests);
+   printf ("| Input Array:            ");
+   DisplayArray (inputs, size);
+   printf ("| Expected Output Array:  ");
+   DisplayArray (expected, size);
+   printf ("| Sorted Output Array:    ");
+   DisplayArray (sortedArray, size);
+   printf ("| Sorting: %s\n", IsSorted (sortedArray, size) ? GREEN "PASS" RESET : RED "FAIL" RESET);
+   if (index >= 0) printf ("Element %d found at sorted index %d.\n", searchElement, index);  // If index >= 0, element was found at that index
+   else printf (RED "Element %d not found.\n" RESET, searchElement);                        // If index == -1, element was not found
+   printf ("--------------------------------------------------\n\n");
+}
+
+/// <summary>Tests sorting and searching algorithms with predefined cases.</summary>
+static void TestSortAndSearch () {
+   struct {
+      int Input[50], Expected[50], SearchElement, Size;
+   } tests[] = {
+       {{34, 7, 23, 32, 10}, {7, 10, 23, 32, 34}, 2, 5},          // Search for an element not in the array
+       {{5, 3, 8, 1, 2, 6}, {1, 2, 3, 5, 6, 8}, 8, 6},
+       {{11, 2}, {2, 11}, 11, 2},
+       {{12, 11, 13, 5}, {5, 11, 12, 13}, 5, 4},
+       {{-1, -2, -3, -4, -5, -6, -7}, {-7, -6, -5, -4, -3, -2, -1}, -3, 7},
+       {{99, 3, 14}, {3, 14, 99}, 14, 3},
+       {{-10, 0, 5, 2}, {-10, 0, 2, 5}, 0, 4},
+       {{1, 2, 3, 4, 5}, {1, 2, 3, 4, 5}, 6, 5}                   // Search for an element not in the array
+   };
+   int numTests = sizeof (tests) / sizeof (tests[0]);
+   for (int i = 0; i < numTests; i++) {
+      int sortedArray[50];
+      for (int j = 0; j < tests[i].Size; j++) sortedArray[j] = tests[i].Input[j];

```

Comment:
This copying into a new array is not required. You can directly pass tests[i] to your sort functionn.